### DISCUSSION TITLE
Preventing Anonymous Key Access to Edge Functions

### DISCUSSION THREAD

**Discussion by ChuckJonas** [supabase/supabase#16674] [Questions]
Right now, the edge functions are essentially "open by default" because the anonymous key satisfies the JWT check (similar to #4547 but for the edge functions)  

Beyond performing a custom code check in every function, is there some way to prevent the anonymous key from accessing Edge Endpoints?

**Comment by SuperDino484** [comment]
Have you found an answer?

I had to do this to check if the key represents an actual user.
```typescript
const supabaseClient = createClient(
    Deno.env.get("SUPABASE_URL") ?? "",
    Deno.env.get("SUPABASE_ANON_KEY") ?? "",
  );

  const {data: {user}} = await supabaseClient.auth.getUser(
    req.headers.get('Authorization')?.replace('Bearer ', '')
  )

  if(user === null) {
    throw new Response(JSON.stringify({ error: "Must be logged in!" }), {
      headers: { ...corsHeaders, "Content-Type": "application/json" },
      status: 401,
    });
  }
